src/models/device_model.jl

```diff
@@ -1,143 +1,136 @@
+    voltage_r,
+    voltage_i,
+    current_r,
+    current_i,
+    device::PSY.PowerLoad,
+    sys::PSY.System,
+)
 
     mdl_Zload!(voltage_r, voltage_i, current_r, current_i, device, sys)
 
     return
 end
 
-function device_model!(x,
-                        output_ode::Vector{Float64},
-                        voltage_r,
-                        voltage_i,
-                        current_r,
-                        current_i,
-                        ix_range::UnitRange{Int64},
-                        ode_range::UnitRange{Int64},
-                        controls,
-                        device::DynI,
-                        sys::DynamicSystem) where {DynI <: DynInverter}
-
-        #Obtain local device states
-        n_states = total_device_states(device)
-        device_states = @view x[ix_range]
-
-        #Obtain references
-        Sbase = get_sys_base(sys)
-        sys_f = get_sys_f(sys)
-
-        #Update Voltage data
-        device.inner_vars[VR_inv_var] = voltage_r[1]
-        device.inner_vars[VI_inv_var] = voltage_i[1]
-        #device.inner_vars[Vh_var] = sqrt(voltage_r[1]^2 + voltage_i[1]^2)
-
-        #Update V_ref
-        V_ref = get_inverter_Vref(device)
-        device.inner_vars[v_control_var] = V_ref
-
-        #Obtain ODES for DC side
-        mdl_DCside_ode!(device)
-
-        #Obtain ODEs for OuterLoop
-        mdl_outer_ode!(device_states,
-                       view(output_ode, ode_range),
-                       sys_f,
-                       device)
-
-#Obtain ODEs for PLL
-mdl_freq_estimator_ode!(device_states,
-                        view(output_ode, ode_range),
-                        sys_f,
-                        device)
-
-#Obtain inner controller ODEs and modulation commands
-mdl_VScontrol_ode!(device_states,
-                   view(output_ode, ode_range),
-                   device)
-
-#Obtain converter relations
-mdl_converter_ode!(device)
-
-#Obtain ODEs for output filter
-mdl_filter_ode!(device_states,
-                        view(output_ode, ode_range),
-                        current_r,
-                        current_i,
-                        Sbase,
-                        sys_f,
-                        device)
-
-
-
-
-        return
+function device_model!(
+    x,
+    output_ode::Vector{T},
+    voltage_r,
+    voltage_i,
+    current_r,
+    current_i,
+    ix_range::UnitRange{Int64},
+    ode_range::UnitRange{Int64},
+    device::DynI,
+    sys::PSY.System,
+) where {DynI<:PSY.DynamicInverter,T<:Real}
+
+    #Obtain local device states
+    n_states = PSY.get_n_states(device)
+    device_states = @view x[ix_range]
+
+    #Obtain references
+    Sbase = PSY.get_basepower(sys)
+    sys_f = PSY.get_frequency(sys)
+
+    #Update Voltage data
+    get_inner_vars(device)[VR_inv_var] = voltage_r[1]
+    get_inner_vars(device)[VI_inv_var] = voltage_i[1]
+
+    #Update V_ref
+    V_ref = PSY.get_ext(device)[CONTROL_REFS][V_ref_index]
+    get_inner_vars(device)[v_control_var] = V_ref
```

Comment:
This looks weird, needs some rename to make it clear if this is the device setpoint or other variable

Reply:
> This looks weird, needs some rename to make it clear if this is the device setpoint or other variable @rodrigomha

I added a new issue to update both struct names and inner vars ref for the inverter. Probably will do before merging to 0.3 or for 0.4 when we add grid following and forming.